src/main/java/org/jabref/logic/importer/CompositeIdFetcher.java

```diff
@@ -21,7 +21,7 @@ public CompositeIdFetcher(ImportFormatPreferences importFormatPreferences) {
     }
 
     public Optional<BibEntry> performSearchById(String identifier) throws FetcherException {
-        Optional<DOI> doi = DOI.parse(identifier);
+        Optional<DOI> doi = DOI.findInText(identifier);
```

Comment:
Semantically, you would want to parse the identifier here and not find it in some bigger text. Thus, if you think "findInText" is better, then please change the implementation of "parse". This makes it also more coherent with the other id fetchers below that use "parse" as well (and the idea was to extract this to a common interface at some point).

Reply:
I am working on the parser in the PR #8228

Reply:
I use single regexp to parse the DOI in and added several test cases in DOITest PR #8228. Also address the request by @koppor for informing the user that "no DOI data exists" rather than "invalid DOI" when the fetcher fails to download from a valid DOI (#8217).

Reply:
I prefer the approach by @mrcstan in #8228.

Reply:
> I use single regexp to parse the DOI in and added several test cases in DOITest PR #8228. Also address the request by @koppor for informing the user that "no DOI data exists" rather than "invalid DOI" when the fetcher fails to download from a valid DOI (#8217).

i think you meant to reference #8127, not 8217. I just got confused when i followed the link in 8217 to this thread. This one here deals with an entirely different topic.